feat: integrate premake manager cli

lolrobbe2 opened the pull request:
@ColtMcG1 i implemented premake manager cli.
this PR contains all the neccesary changes to implement premake manager.

# commands:

## version

1) ListAvailable
2) ListInstalled
3) Set

## Module

1) Add
2) Info
3) Install

## Config

1) View
2) version

lolrobbe2 committed:
feat: premake terminal

lolrobbe2 committed:
feat: added list releases command

lolrobbe2 committed:
feat: added multiple new commands

lolrobbe2 committed:
feat: added module commands

ColtMcG1 commented:
@lolrobbe2 Any thoughts on Copilots recommendations? Its pretty late for me atm. Let me just do a once over of the PR tomorrow before I merge.

lolrobbe2 committed:
fix: implemented copilot suggestions

ColtMcG1 commented:
@lolrobbe2 I'm getting a missing file error if I try any other command besides the module commands. Will work on troubleshooting the error.

ColtMcG1 commented:
1. Looks like the code that looks for Premake manager in the temp directory wouldn't write it if it didn't exist.

2. I am wondering if you would be able to make a build of Premake manager that would function as a CLI server? 
That way we could launch the CLI when the extension starts and leave it run. Inject commands on stdin, and have the CLI reply on stdout. That way a new process isn't spun up every time a command is executed.
Just a thought. The dev time may not be worth the performance gain.

3. Also, when I am running commands, Premake manager is returning an error about a missing directory in the roaming temp folder. Might you be able to check that out?

lolrobbe2 commented:
@ColtMcG1 

1) good catch
2) i havemt heard of CLI servers, and this will make it easier to implement for tother extension playforms
3) could you pls send a copy of the error, ill look into it. (It might be the fact that premake is not installed in the folder)

colton-mcgraw committed:
Fix bug in logic that would not write the premake-manger exe to temp folder if it didn't exist.

ColtMcG1 commented:
Here is an overview of how a  CLI server for IPC (intra process communication) works.

## Google Summary
For true IPC, you can pipe a file or another process's output into your server's standard input. This simulates how a GUI or another application would use it.

1. Create a text file named commands.txt with the following content:
```text
ECHO Hello from a file!
REVERSE File input test
EXIT
```

2. Run the server and redirect the file's content to its input:
```sh
dotnet run < commands.txt
```

3. The output will be processed and printed to the terminal:
```text
CLI server started. Awaiting commands...
Hello from a file!
tset tupni eliF
Shutting down.
```

How the server works

Continuous loop: The while (true) loop keeps the program running and listening for commands.

Standard input (Console.ReadLine): The program *waits* for a line of text to be entered. This can be from the keyboard, a piped file, or another process. The loop terminates gracefully if null is returned, which happens when a redirected input stream ends.

Command parsing: The input string is split to extract a command and any arguments. Using .ToUpper() on the command makes the server case-insensitive.
Standard output (Console.WriteLine): Responses are written back to the standard output stream, where they can be read by a human user or another program.
Standard error (Console.Error.WriteLine): Using Console.Error for error messages is a common best practice in CLI tools. It separates diagnostic information from the program's intended output, allowing other applications to filter or ignore errors. 

## My notes:
In my limited experience so far trying to implement this, the best way is to use JSON-over-the-wires and have a strict schema. This way its an API like interface face and each process know what's its expecting to get. Easy to process, safe, and easy to catch errors.

The other options is a gRPC server. But those are a **TON** of work to implement! Not really needed for this use case.

colton-mcgraw committed:
Add `premake-manager-cli` as a Git submodule

A new Git submodule named `premake-manager-cli` has been added to the repository. The submodule is located at the path `premake-manager-cli` and points to the remote repository URL `https://github.com/lolrobbe2/premake-manager-cli.git`.

The submodule is locked to the specific commit `6a4415bf61934434403c3d535eff2509f206ffa0` to ensure consistency in the version being used.
